mutt/file.c

```diff
@@ -561,6 +562,17 @@ int mutt_file_open(const char *path, uint32_t flags)
   struct Buffer safe_file = buf_make(0);
   struct Buffer safe_dir = buf_make(0);
 
+  mode_t current_mask;
+  if (cs_subset_bool(NeoMutt->sub, "attachments_respect_umask"))
```

Comment:
`mutt_file_open` is used for everything - not just attachments. So we either need to make the config option "global" or we need to make this change only for attachments.